### PR TITLE
feat: adding JSON scalar

### PR DESCRIPTION
**Pull request by mnorbury**
fixes: #63

### CODE REVIEW

File: lib/src/main/java/io/cloudquery/scalar/JSON.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package io.cloudquery.scalar;
+
+import static io.cloudquery.scalar.ValidationException.NO_CONVERSION_AVAILABLE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (erezrokah) [inline lib/src/main/java/io/cloudquery/scalar/JSON.java:5]: Should `jackson` be added as a dependency to `build.gradle`?

Unrelated, but I used `gson` in https://github.com/cloudquery/plugin-sdk-java/pull/81/files#diff-655a69127303f6948c0b150902436756156ec7f82640e994c1f552cbdec5bbceR48

File: lib/src/main/java/io/cloudquery/scalar/JSON.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      if (bytes.length == 0) {
+        return;
+      }
+      if (!isValidJSON(bytes)) {
+        throw new ValidationException("invalid json", dt, value);
+      }
+      this.value = bytes;
+    } else if (value instanceof java.lang.String string) {
+      set(string.getBytes());
+    } else {
```
Comment (erezrokah) [inline lib/src/main/java/io/cloudquery/scalar/JSON.java:36]: We should also allow setting an instance of `JSON` (the scalar `JSON`). We can check if the value is an instance of the class, then set the underlying value directly

**Review by erezrokah — CHANGES_REQUESTED** [review]
Looks great. A couple comments. Once this is merge I'll update my PR to use Jackson

File: lib/build.gradle [hunk 3]
```diff
@@ -42,6 +42,9 @@ dependencies {
     implementation "org.apache.arrow:arrow-memory-core:12.0.1"
     implementation "org.apache.arrow:arrow-vector:12.0.1"
 
+    implementation "com.fasterxml.jackson.core:jackson-annotations:2.15.1"
+    implementation "com.fasterxml.jackson.core:jackson-annotations:2.15.1"
```
Comment (erezrokah) [inline lib/build.gradle:46]:
```suggestion
```

File: lib/src/main/java/io/cloudquery/scalar/JSON.java (hunk 1 above)
Comment (mnorbury) [reply lib/src/main/java/io/cloudquery/scalar/JSON.java:5]: Do you have a preference. I think I saw some hints that Jackson is slightly faster, which is why I went with it, but I have no solid preference.

File: lib/src/main/java/io/cloudquery/scalar/JSON.java (hunk 2 above)
Comment (mnorbury) [reply lib/src/main/java/io/cloudquery/scalar/JSON.java:36]: Actually, I think this scenario is covered in the abstract class [here](https://github.com/cloudquery/plugin-sdk-java/blob/1d31031b08d4784b54dc2a589d9c3b76d3e4f312/lib/src/main/java/io/cloudquery/scalar/Scalar.java#L37-L41)

Comment (erezrokah) [reply lib/src/main/java/io/cloudquery/scalar/JSON.java:36]: Ah nice, didn't see that. Nice one

File: lib/build.gradle (hunk 3 above)
Comment (mnorbury) [reply lib/build.gradle:46]: The second dependency was supposed to be the `jackson-core` fixed [here](https://github.com/cloudquery/plugin-sdk-java/pull/82/commits/a8b12bd4b776b6e3cb1dd562770bac17ac80ab50)

**Review by erezrokah — APPROVED** [review]

File: lib/build.gradle (hunk 3 above)
Comment (erezrokah) [reply lib/build.gradle]: Woops deleted one by mistake, let me bring it back

Comment (erezrokah) [reply lib/build.gradle]: Actually we don't need annotations at the moment, correct?

Comment (mnorbury) [reply lib/build.gradle]: I did add them [here](https://github.com/cloudquery/plugin-sdk-java/blob/1d31031b08d4784b54dc2a589d9c3b76d3e4f312/lib/src/main/java/io/cloudquery/transformers/NameTransformer.java#L23) in a previous PR, so we do support them for custom JSON field names.

Comment (erezrokah) [reply lib/build.gradle]: Ah, cool. Let me add it back

**Pull request merged**